src/core/symbology/qgsfillsymbollayer.cpp

```diff
@@ -4132,12 +4134,14 @@ void QgsPointPatternFillSymbolLayer::renderPolygon( const QPolygonF &points, con
     case Qgis::MarkerClipMode::CompletelyWithin:
     {
       shapePolygon = std::make_unique< QgsPolygon >();
-      shapePolygon->setExteriorRing( QgsLineString::fromQPolygonF( points ) );
+      std::unique_ptr< QgsLineString > fromPolygon = QgsLineString::fromQPolygonF( points );
+      shapePolygon->setExteriorRing( fromPolygon.release() );
```

Comment:
shouldn't we wait to have a variant of setExteriorRing() that accepts a unique_ptr to avoid the release() ?  (I know the topic is dependent on PyQt)

Reply:
> shouldn't we wait to have a variant of setExteriorRing() that accepts a unique_ptr to avoid the release() ? (I know the topic is dependent on PyQt)

It's dependent on PyQt, and I honestly don't see it as possible without changes in sip itself. And then we'd need to wait YEARS for that to be available everywhere we need...